### PR TITLE
[WasmFS] Implement chmod syscall

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

File: tests/unistd/unlink.c [hunk 1]
```diff
@@ -150,7 +136,7 @@ void test() {
   err = access("file1", F_OK);
   assert(!err);
 #endif
-#ifndef NO_SYMLINK
+#if !defined(NO_SYMLINK) && !defined(WASMFS)
```
Comment (sbc100) [inline tests/unistd/unlink.c:139]: Add a TODO with a link to a bug maybe?

**Review by sbc100 — APPROVED** [review]

File: tests/unistd/unlink.c (hunk 1 above)
Comment (kripken) [reply tests/unistd/unlink.c:139]: I already have the PR to remove this all ready to go... :smile:

File: tests/test_core.py [hunk 2]
```diff
@@ -5650,8 +5653,10 @@ def test_unistd_unlink(self):
       if fs == 'WASMFS':
```
Comment (tlively) [inline tests/test_core.py:5653]: I believe this is now dead code. Relatedly, there is a lot of setup logic here that is not directly related to unlinking. Is that logic going to be repeated for multiple tests? It might be worth abstracting away if so.

Comment (kripken) [reply tests/test_core.py]: Good catch, removed the dead code now.

I'm not sure if this setup code repeats in other tests. If I see it again in another test that I get to later I'll keep a lookup for refactoring opportunities.

**Review by tlively — APPROVED** [review]
